src/psyclone/psyir/backend/fortran.py

```diff
@@ -1640,6 +1640,12 @@ def call_node(self, node):
             else:
                 result_list.append(self._visit(child))
         args = ", ".join(result_list)
+        if isinstance(node, IntrinsicCall):
+            if node.routine.name in ["ALLOCATE", "DEALLOCATE"]:
+                # An allocate/deallocate doesn't have 'call'.
+                return f"{self._nindent}{node.routine.name}({args})\n"
+            if node.routine.name == "RANDOM":
+                return f"{self._nindent}call RANDOM_NUMBER({args})\n"
```

Comment:
I prefer if you rename the PSyIR RANDOM to RANDOM_NUMBER (to not confuse them with other fuctions, like the intel specific https://www.intel.com/content/www/us/en/develop/documentation/fortran-compiler-oneapi-dev-guide-and-reference/top/language-reference/a-to-z-reference/q-to-r/rand-random.html) and then update the backend as:
```
else:
    return f"{self._nindent}call {node.routine.name}({args})\n"
```

Reply:
Or maybe remove the second condition... the rest of the visitor will already do the right thing, its just ALLOCATE, DEALOCATE that are exceptions.

Reply:
Yes, I've simplified.